packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworking.tsx

```diff
@@ -229,10 +244,69 @@ class LinodeNetworking extends React.Component<CombinedProps, State> {
           initialLoading: false,
         });
       });
+
+    const refreshIPv6 = getAllIPv6Ranges({})
+      .then(async (resp) => {
+        const ranges = resp.data;
+
+        const sharedRanges: IPRange[] = [];
+        const staticRanges: IPRange[] = [];
+        const availableRanges: IPRangeInformation[] = [];
+        const rangeConstruction = await ranges.reduce(async (acc, range) => {
+          await acc;
+          // filter user ranges outside dc
+          if (range.region !== this.props.linode.region) {
+            return acc;
+          }
+
+          await getIPv6RangeInfo(range.range).then(async (resp) => {
+            let slaac;
+            if (!resp.is_bgp) {
+              await getLinodeIPs(this.props.linode.id).then((ips) => {
+                slaac = ips?.ipv6?.slaac.address;
+              });
+            }
+
+            if (resp.is_bgp && resp.linodes.includes(this.props.linode.id)) {
+              // any range that is shared to this linode
+              this.props.flags.ipv6Sharing && sharedRanges.push(range);
+            } else if (!resp.is_bgp && range.route_target === slaac) {
+              // any range that is statically routed to this linode
+              staticRanges.push(range);
+            } else {
+              // any range that is not shared to this linode or static on this linode
+              this.props.flags.ipv6Sharing && availableRanges.push(resp);
+            }
```

Comment:
Can we move the flag checking to be the first thing checked for short circuiting?

```suggestion
            if (
              this.props.flags.ipv6Sharing &&
              resp.is_bgp &&
              resp.linodes.includes(this.props.linode.id)
            ) {
              // any range that is shared to this linode
              sharedRanges.push(range);
            } else if (!resp.is_bgp && range.route_target === slaac) {
              // any range that is statically routed to this linode
              staticRanges.push(range);
            } else if (this.props.flags.ipv6Sharing) {
              // any range that is not shared to this linode or static on this linode
              availableRanges.push(resp);
            }
```